File: src/dataset/typescript/twelve-days_exercism-wackerow/claude/ClaudeSonnet4_twelve-days_v1.ts
class TwelveDays {
    private static readonly _numbers = [
        'first',
        'second',
        'third',
        'fourth',
        'fifth',
        'sixth',
        'seventh',
        'eighth',
        'ninth',
        'tenth',
        'eleventh',
        'twelfth',
    ]

    private static readonly _gifts = [
        'a Partridge in a Pear Tree.\n',
        'two Turtle Doves, ',
        'three French Hens, ',
        'four Calling Birds, ',
        'five Gold Rings, ',
        'six Geese-a-Laying, ',
        'seven Swans-a-Swimming, ',
        'eight Maids-a-Milking, ',
        'nine Ladies Dancing, ',
        'ten Lords-a-Leaping, ',
        'eleven Pipers Piping, ',
        'twelve Drummers Drumming, '
    ]

    private static readonly _andGift = 'and a Partridge in a Pear Tree.\n'

    public static recite(fromVerse: number, toVerse: number): string {
        const verses: string[] = []
        
        for (let i = fromVerse - 1; i < toVerse; i++) {
            const gifts: string[] = []
            
            for (let j = i; j > 0; j--) {
                gifts.push(this._gifts[j])
            }
            
            if (i > 0) {
                gifts.push(this._andGift)
            } else {
                gifts.push(this._gifts[0])
            }
            
            verses.push(`On the ${this._numbers[i]} day of Christmas my true love gave to me: ${gifts.join('')}`)
        }
        
        return verses.join('')
    }
}

export default TwelveDays

File: src/dataset/Java/twelve-days_exercism-uzilan/claude/ClaudeSonnet4_TwelveDays_v1.java
import java.util.Arrays;
import java.util.List;

class TwelveDays {

    String verse(int verseNumber) {
        final StringBuilder sb = new StringBuilder(150); // Pre-allocate capacity
        sb.append("On the ").append(days[verseNumber - 1]).append(" day of Christmas my true love gave to me, ");

        for (int i = verseNumber - 1; i > 0; i--) {
            sb.append(gifts[i]);
        }

        if (verseNumber > 1) {
            sb.append("and ");
        }

        sb.append(gifts[0]).append("\n");

        return sb.toString();
    }

    String verses(int startVerse, int endVerse) {
        final StringBuilder sb = new StringBuilder((endVerse - startVerse + 1) * 200); // Pre-allocate capacity
        
        for (int i = startVerse; i <= endVerse; i++) {
            sb.append(verse(i));
            if (i < endVerse) {
                sb.append("\n");
            }
        }
        
        return sb.toString();
    }

    String sing() {
        return verses(1, days.length);
    }

    private static final String[] days = {
            "first",
            "second",
            "third",
            "fourth",
            "fifth",
            "sixth",
            "seventh",
            "eighth",
            "ninth",
            "tenth",
            "eleventh",
            "twelfth"};

    private static final String[] gifts = {
            "a Partridge in a Pear Tree.",
            "two Turtle Doves, ",
            "three French Hens, ",
            "four Calling Birds, ",
            "five Gold Rings, ",
            "six Geese-a-Laying, ",
            "seven Swans-a-Swimming, ",
            "eight Maids-a-Milking, ",
            "nine Ladies Dancing, ",
            "ten Lords-a-Leaping, ",
            "eleven Pipers Piping, ",
            "twelve Drummers Drumming, "};
}